proposed/swift-interop.md

```diff
@@ -152,33 +152,40 @@ If possible, Swift tuples should be represented as `ValueTuple`s in .NET. If thi
 
 #### Projection Tooling Components
 
-The projection tooling should be split into these components:
+The projection tooling will be based on the [Binding Tools for Swift](https://github.com/xamarin/binding-tools-for-swift) (BTfS). The BTfS contains a set of tools that can consume a compiled Apple Swift library and generate wrappers (bindings) that allow it to be surfaced as a .NET library. This tool will be implemented as a self-hosted .NET CLI tool, automating the generation of wrappers by parsing the Swift library interface. It generates C# bindings that can be utilized as a .NET library and Swift bindings that C# bindings invoke in cases where direct P/Invoke into Swift library is not possible.
+
+The projection tooling should be split into these components.
 
 ##### Importing Swift into .NET
 
-1. A tool that takes in a `.swiftinterface` file or Swift sources and produces C# code.
-2. A library that provides the basic support for Swift interop that the generated code builds on.
-3. User tooling to easily generate Swift projections for a given set of `.framework`s.
-    - This tooling would build a higher-level interface on top of the tool in item 1 that is more user-friendly and project-system-integrated.
-4. (optional) A NuGet package, possibly referencable by `FrameworkReference` or automatically included when targeting macOS, Mac Catalyst, iOS, or tvOS platforms that exposes the platform APIs for each `.framework` that is exposed from Swift to .NET.
-    - This would be required to provide a single source of truth for Swift types so they can be exposed across an assembly boundary.
+Importing Swift into .NET is done through the following steps:
+
+1. The tool analyzes a source `.swift` library and maps entry points to mangled names.
+2. Then,the tool aggregates the public API; it takes in a `.swiftmodule` file (similar to a C header file) generated by the Swift compiler, creating a module declaration.
+3. For cases where direct P/Invoke from C# into Swift is not possible, the tool generates `.swift` source code wrappers along with a `.dylib` compiled native library.
```

Comment:
What do we expect these cases to be? I though there should be none.

Reply:
Here are some examples:
 - Passing a struct by value in more registers than P/Invoke will allow -- requires "unpacking" from Swift side
 - Exporting .NET into Swift -- open classes with virtual members and simulated vtables
 - Closures

/cc: @stephen-hawley

Reply:
> Passing a struct by value in more registers than P/Invoke will allow -- requires "unpacking" from Swift side

This does not sound right. I thought that the our goal with CallConvSwift design was that we should be able to call any signature, without a wrapper on the Swift side.

> Exporting .NET into Swift

This paragraph talks about importing .NET into Swift...

Reply:
Your concerns are valid. Ideally, we should aim to eliminate Swift wrappers and reduce C# bindings to a minimum. I suggest we move towards this goal iteratively and have a lean approach with the first version of the interop.

The cases where direct P/Invoke from C# into Swift is not possible include types that don't have one-to-one mapping with Swift, including some edge cases like mentioned structs. I recommend listing these limitations in this PR and subsequent PRs.

Regarding exporting .NET into Swift, I believe the problem is more complex, but it's not the main scenario for the interop.

Reply:
Updated the proposal to reflect the intended behavior instead of the current state. Resolving this comment in favor of https://github.com/dotnet/designs/pull/312#discussion_r1453961718.